gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/query_service/media/MediaQueryServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.infra.query_service.media;
+
+import io.gravitee.apim.core.media.model.Media;
+import io.gravitee.apim.core.media.query_service.MediaQueryService;
+import io.gravitee.apim.infra.adapter.GraviteeDefinitionAdapter;
+import io.gravitee.repository.exceptions.TechnicalException;
+import io.gravitee.repository.media.api.MediaRepository;
+import io.gravitee.rest.api.service.exceptions.TechnicalManagementException;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class MediaQueryServiceImpl implements MediaQueryService {
+
+    private final MediaRepository mediaRepository;
+
+    @Override
+    public List<Media> findAllByApiId(String apiId) {
+        try {
+            return GraviteeDefinitionAdapter.INSTANCE.mapMedia(mediaRepository.findAllByApi(apiId));
```

Comment:
What about creating a MediaAdapter? It is strange to use the GraviteeDefinitionAdapter to adapt a Media 🤔 
Usually for other Query/Crud service we have a "dedicated" adapter to handle the mapping